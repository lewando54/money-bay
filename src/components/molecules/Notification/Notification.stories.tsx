/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import Notification from './Notification'

const meta = {
	component: Notification,
	argTypes: {
		type: { control: 'select', options: ['error', 'warning', 'correct'] },
		title: { control: 'text' },
		body: { control: 'text' },
		datetime: { control: 'date' }
	}
} satisfies Meta<typeof Notification>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
	args: {
		type: 'correct',
		title: 'Your loan application has been approved!',
		body: 'Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
		datetime: new Date('2022-09-12T12:32:12')
	}
}
