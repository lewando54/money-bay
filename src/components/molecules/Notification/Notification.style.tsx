import { StyleSheet } from 'react-native'

import globalThemeStyle, { BODY_TEXT_COLOR, MAIN_DARK } from '@styling/GlobalTheme.style'
export default StyleSheet.create({
    iconAndTitleWrapper: {
        flexDirection: 'row', 
        gap: 8, 
        alignItems: 'flex-start'
    },
    titleWrapper: {
        alignItems: 'flex-start',
        flexWrap: 'wrap',
        width: '100%',
        flexShrink: 1,
        marginTop: -3
    },
    notificationTitleText: {
        ...globalThemeStyle.text_SemiBold,
        fontSize: 16,
        color: MAIN_DARK
    },
    notificationBodyText: {
        ...globalThemeStyle.text_Regular, 
        fontSize: 16,
        color: MAIN_DARK
    },
    notificationTimeText: {
        ...globalThemeStyle.text_Regular,
        fontSize: 12,
        color: BODY_TEXT_COLOR
    }
})
