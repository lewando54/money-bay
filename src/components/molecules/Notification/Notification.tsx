import { View, Text } from 'react-native'
import UniversalContainer from '../../atoms/UniversalContainer/UniversalContainer'
import notificationStyle from './Notification.style'
import CircleWithIcon from '../../atoms/CircleWithIcon/CircleWithIcon'
import { MaterialIcons } from '@expo/vector-icons';
import { AntDesign } from '@expo/vector-icons'; 
import globalThemeStyle, { BODY_TEXT_COLOR, GOOD_GREEN, LINK_COLOR, MAIN_DARK, WARNING_YELLOW } from '../../../styling/GlobalTheme.style';
import Separator from '../../atoms/Separator/Separator';

type TNotificationType = 'warning' | 'error' | 'correct'

interface INotificationProps {
    type?: TNotificationType
    title?: string
    body?: string
    datetime?: Date
    testID?: string
}

export default function Notification({
    type='correct',
    title='Your loan application has been approved!',
    body='Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
    datetime=new Date('2022-09-12T12:32:12'),
    testID
}: INotificationProps){
    const notificationTypeIcon = 
        type == 'correct' ? 
            <CircleWithIcon backgroundColor={GOOD_GREEN} size={18} icon={<MaterialIcons name="check" size={10} color="white" />}/> :
        type == 'error' ?
            <CircleWithIcon backgroundColor={LINK_COLOR} size={18} icon={<MaterialIcons name="close" size={10} color="white" />}/> :
        type == 'warning' &&
            <CircleWithIcon backgroundColor={WARNING_YELLOW} size={18} icon={<AntDesign name="warning" size={10} color="white" />}/>

    return (
        <UniversalContainer padding={22} style={{gap: 12}} testID={testID}>
            <View style={notificationStyle.iconAndTitleWrapper}>
                {notificationTypeIcon}
                <View style={notificationStyle.titleWrapper}>
                    <Text style={[globalThemeStyle.text_SemiBold, { fontSize: 16, color: MAIN_DARK }]}>{title}</Text>
                </View>
            </View>
            <Text style={[globalThemeStyle.text_Regular, { fontSize: 16, color: MAIN_DARK }]}>{body}</Text>
            <Separator />
            <Text style={[globalThemeStyle.text_Regular, { fontSize: 12, color: BODY_TEXT_COLOR }]}>{datetime.toLocaleDateString(undefined, {year: 'numeric', month: 'short', day: 'numeric'})} at {datetime.toLocaleTimeString(undefined, {hour: 'numeric', minute: 'numeric'})}</Text>
        </UniversalContainer>
    )
}
