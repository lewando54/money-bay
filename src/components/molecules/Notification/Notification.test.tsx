import React from 'react'
import renderer from 'react-test-renderer'

import Notification from './Notification'
describe('Notification', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<Notification />)
        expect(tree).toMatchSnapshot()
    })

    it('should render green icon with checkmark by default', () => {
        const tree = renderer.create(<Notification type={'correct'}/>)
        expect(tree).toMatchSnapshot()
    })

    it('should render green icon with checkmark', () => {
        const tree = renderer.create(<Notification type='correct'/>)
        expect(tree).toMatchSnapshot()
    })

    it('should render yellow/orange icon with warning sign', () => {
        const tree = renderer.create(<Notification type='warning'/>)
        expect(tree).toMatchSnapshot()
    })

    it('should render red icon with x', () => {
        const tree = renderer.create(<Notification type='error'/>)
        expect(tree).toMatchSnapshot()
    })
})
