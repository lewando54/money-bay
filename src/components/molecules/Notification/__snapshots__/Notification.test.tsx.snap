// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should render correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 22,
        "width": "100%",
      },
      false,
      {
        "gap": 12,
      },
    ]
  }
  testID="universal-container"
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#3EB290",
            "borderRadius": 36,
            "height": 18,
            "justifyContent": "center",
            "width": 18,
          },
          false,
        ]
      }
    >
      <Text />
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexShrink": 1,
          "flexWrap": "wrap",
          "marginTop": -3,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 16,
          }
        }
      >
        Your loan application has been approved!
      </Text>
    </View>
  </View>
  <Text
    style={
      {
        "color": "#1B1D4D",
        "fontFamily": "MulishRegular",
        "fontSize": 16,
      }
    }
  >
    Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#CED6E1",
          "height": 1,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <Text
    style={
      {
        "color": "#4C4C60",
        "fontFamily": "MulishRegular",
        "fontSize": 12,
      }
    }
  >
    Sep 12, 2022
     at 
    12:32 PM
  </Text>
</View>
`;

exports[`Notification should render green icon with checkmark 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 22,
        "width": "100%",
      },
      false,
      {
        "gap": 12,
      },
    ]
  }
  testID="universal-container"
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#3EB290",
            "borderRadius": 36,
            "height": 18,
            "justifyContent": "center",
            "width": 18,
          },
          false,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "white",
              "fontSize": 10,
            },
            undefined,
            {
              "fontFamily": "material",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexShrink": 1,
          "flexWrap": "wrap",
          "marginTop": -3,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 16,
          }
        }
      >
        Your loan application has been approved!
      </Text>
    </View>
  </View>
  <Text
    style={
      {
        "color": "#1B1D4D",
        "fontFamily": "MulishRegular",
        "fontSize": 16,
      }
    }
  >
    Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#CED6E1",
          "height": 1,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <Text
    style={
      {
        "color": "#4C4C60",
        "fontFamily": "MulishRegular",
        "fontSize": 12,
      }
    }
  >
    Sep 12, 2022
     at 
    12:32 PM
  </Text>
</View>
`;

exports[`Notification should render green icon with checkmark by default 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 22,
        "width": "100%",
      },
      false,
      {
        "gap": 12,
      },
    ]
  }
  testID="universal-container"
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#3EB290",
            "borderRadius": 36,
            "height": 18,
            "justifyContent": "center",
            "width": 18,
          },
          false,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "white",
              "fontSize": 10,
            },
            undefined,
            {
              "fontFamily": "material",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexShrink": 1,
          "flexWrap": "wrap",
          "marginTop": -3,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 16,
          }
        }
      >
        Your loan application has been approved!
      </Text>
    </View>
  </View>
  <Text
    style={
      {
        "color": "#1B1D4D",
        "fontFamily": "MulishRegular",
        "fontSize": 16,
      }
    }
  >
    Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#CED6E1",
          "height": 1,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <Text
    style={
      {
        "color": "#4C4C60",
        "fontFamily": "MulishRegular",
        "fontSize": 12,
      }
    }
  >
    Sep 12, 2022
     at 
    12:32 PM
  </Text>
</View>
`;

exports[`Notification should render red icon with x 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 22,
        "width": "100%",
      },
      false,
      {
        "gap": 12,
      },
    ]
  }
  testID="universal-container"
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#FF5887",
            "borderRadius": 36,
            "height": 18,
            "justifyContent": "center",
            "width": 18,
          },
          false,
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "white",
              "fontSize": 10,
            },
            undefined,
            {
              "fontFamily": "material",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexShrink": 1,
          "flexWrap": "wrap",
          "marginTop": -3,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 16,
          }
        }
      >
        Your loan application has been approved!
      </Text>
    </View>
  </View>
  <Text
    style={
      {
        "color": "#1B1D4D",
        "fontFamily": "MulishRegular",
        "fontSize": 16,
      }
    }
  >
    Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#CED6E1",
          "height": 1,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <Text
    style={
      {
        "color": "#4C4C60",
        "fontFamily": "MulishRegular",
        "fontSize": 12,
      }
    }
  >
    Sep 12, 2022
     at 
    12:32 PM
  </Text>
</View>
`;

exports[`Notification should render yellow/orange icon with warning sign 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": "auto",
        "padding": 22,
        "width": "100%",
      },
      false,
      {
        "gap": 12,
      },
    ]
  }
  testID="universal-container"
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "gap": 8,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#EECC55",
            "borderRadius": 36,
            "height": 18,
            "justifyContent": "center",
            "width": 18,
          },
          false,
        ]
      }
    >
      <Text />
    </View>
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexShrink": 1,
          "flexWrap": "wrap",
          "marginTop": -3,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishSemiBold",
            "fontSize": 16,
          }
        }
      >
        Your loan application has been approved!
      </Text>
    </View>
  </View>
  <Text
    style={
      {
        "color": "#1B1D4D",
        "fontFamily": "MulishRegular",
        "fontSize": 16,
      }
    }
  >
    Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#CED6E1",
          "height": 1,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <Text
    style={
      {
        "color": "#4C4C60",
        "fontFamily": "MulishRegular",
        "fontSize": 12,
      }
    }
  >
    Sep 12, 2022
     at 
    12:32 PM
  </Text>
</View>
`;
